src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -33,9 +33,30 @@ public List<BookDto> findAll() {
 
     @Override
     public BookDto findById(long id) {
-        Book book = bookRepository.findById(id).orElseThrow(
+        Book book = findBookByIdOrThrow(id);
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto createBookRequestDto) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Book not found by id: " + id));
+
+        Book book = bookMapper.toModel(createBookRequestDto);
+        book.setId(id);
+        Book savedBook = bookRepository.save(book);
+        return bookMapper.toDto(savedBook);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        Book book = findBookByIdOrThrow(id);
```

Comment:
Better to add separate method something like isExistsById, because you fetch book that you don`t use

Reply:
check updateById as well

Reply:
done